extensions/vault/runtime/src/main/java/io/quarkus/vault/internal/client/VaultClient.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.vault.internal.client;
+
+import io.quarkus.vault.internal.client.dto.VaultAdCreds;
+import io.quarkus.vault.internal.client.dto.VaultDbCreds;
+import io.quarkus.vault.internal.client.dto.VaultKubernetesAuth;
+import io.quarkus.vault.internal.client.dto.VaultKvSecretV1;
+import io.quarkus.vault.internal.client.dto.VaultKvSecretV2;
+import io.quarkus.vault.internal.client.dto.VaultLeasesLookup;
+import io.quarkus.vault.internal.client.dto.VaultLookupSelf;
+import io.quarkus.vault.internal.client.dto.VaultRenewLease;
+import io.quarkus.vault.internal.client.dto.VaultRenewSelf;
+import io.quarkus.vault.internal.client.dto.VaultUserPassAuth;
+
+public interface VaultClient {
+
+    String X_VAULT_TOKEN = "X-Vault-Token";
+    String API_VERSION = "v1";
+
+    VaultUserPassAuth loginUserPass(String user, String password);
+
+    VaultKubernetesAuth loginK8s(String role, String jwt);
```

Comment:
s/K8s/Kubernetes/ Let's try to be consistent everywhere.

Reply:
done